inngest/src/main/kotlin/com/inngest/InngestFunctionConfigBuilder.kt

```diff
@@ -113,12 +113,20 @@ class InngestFunctionConfigBuilder {
         key: String? = null,
         scope: ConcurrencyScope? = null,
     ): InngestFunctionConfigBuilder {
+        when (scope) {
+            ConcurrencyScope.ENVIRONMENT -> if (key == null) throw InngestInvalidConfigurationException("Concurrency key required with environment scope")
+            ConcurrencyScope.ACCOUNT -> if (key == null) throw InngestInvalidConfigurationException("Concurrency key required with account scope")
```

Comment:
the ticket originally said function or env scope but dev server shows this

![Screenshot 2024-09-09 at 8 22 02 AM](https://github.com/user-attachments/assets/7058e189-0f39-4f91-a081-c5fbf050d7bf)
![Screenshot 2024-09-09 at 8 20 49 AM](https://github.com/user-attachments/assets/b2f4dda6-56a2-44fd-ac40-8e3d0c0227da)

Reply:
i see yeah that makes sense, the js sdk isn't preventing this either but it could probably prevent it better with a union type instead of exceptions.

Reply:
yeah I was considering a similar idea but it seemed more cumbersome to do in kotlin, so I went with the validation method